stdlib/public/runtime/Errors.cpp

```diff
@@ -254,47 +259,34 @@ void swift::printCurrentBacktrace(unsigned framesToSkip) {
     fprintf(stderr, "<backtrace unavailable>\n");
 }
 
-#ifdef SWIFT_HAVE_CRASHREPORTERCLIENT
-#include <malloc/malloc.h>
-
-// Instead of linking to CrashReporterClient.a (because it complicates the
-// build system), define the only symbol from that static archive ourselves.
-//
-// The layout of this struct is CrashReporter ABI, so there are no ABI concerns
-// here.
-extern "C" {
-SWIFT_LIBRARY_VISIBILITY
-struct crashreporter_annotations_t gCRAnnotations
-__attribute__((__section__("__DATA," CRASHREPORTER_ANNOTATIONS_SECTION))) = {
-    CRASHREPORTER_ANNOTATIONS_VERSION, 0, 0, 0, 0, 0, 0, 0};
-}
-#endif // SWIFT_HAVE_CRASHREPORTERCLIENT
-
 // Report a message to any forthcoming crash log.
 static void
 reportOnCrash(uint32_t flags, const char *message)
 {
 #ifdef SWIFT_HAVE_CRASHREPORTERCLIENT
-  // We must use an "unsafe" mutex in this pathway since the normal "safe"
-  // mutex calls fatalError when an error is detected and fatalError ends up
-  // calling us. In other words we could get infinite recursion if the
-  // mutex errors.
-  static swift::StaticUnsafeMutex crashlogLock;
-
-  crashlogLock.lock();
-
-  char *oldMessage = (char *)CRGetCrashLogMessage();
-  char *newMessage;
-  if (oldMessage) {
-    swift_asprintf(&newMessage, "%s%s", oldMessage, message);
-    if (malloc_size(oldMessage)) free(oldMessage);
-  } else {
-    newMessage = strdup(message);
-  }
-  
-  CRSetCrashLogMessage(newMessage);
+  char *oldMessage = nullptr;
+  char *newMessage = nullptr;
 
-  crashlogLock.unlock();
+  oldMessage = std::atomic_load_explicit(
+    (volatile std::atomic<char *> *)&gCRAnnotations.message,
+    std::memory_order_relaxed);
```

Comment:
Not that it's going to matter much in practice, but this should be `SWIFT_MEMORY_ORDER_CONSUME`.